### PR TITLE
netsuite connector page

### PR DESCRIPTION
**Pull request by NileshPant1999**
## Summary by CodeRabbit

- **New Features**
	- Introduced models for handling Netsuite credentials and subsidiary mappings.
	- Added a `NetsuiteConnectorModel` class for constructing Netsuite connector payloads.
	- Implemented a `NetsuiteOnboardingModel` class to manage Netsuite onboarding steps.
	- New services for managing Netsuite connections, mappings, and workspace interactions.
	- Enhanced the Netsuite onboarding connector component to include onboarding steps and connection status.
	- Introduced forms for connecting to a NetSuite account with input validation.
	- Added styling for the NetSuite connector component.
- **Refactor**
	- Updated Netsuite and Sage300 modules to include necessary imports and remove unused ones.
- **Documentation**
	- Provided detailed comments in new files to explain the purpose and functionality of each component and service.

### CODE REVIEW

File: src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+              <input placeholder="Enter Company ID" type="text" id="tokenID" name="tokenID" formControlName="tokenId" class="custom-focus tw-border tw-border-gray-300 tw-rounded tw-px-3 tw-py-2 tw-w-400-px" [ngClass]="[connectNetsuiteForm.controls.tokenId.invalid && connectNetsuiteForm.controls.tokenId.touched ? 'error-box' : 'normal-box']">
+              <div *ngIf="connectNetsuiteForm.controls.tokenId.invalid && connectNetsuiteForm.controls.tokenId.touched" class="incorrect-credential">
+                Invalid Token ID. Please enter valid Token ID.
+              </div>
+            </div>
+            <div class="tw-mb-4">
+              <p for="userPassword" class="tw-text-faded-text-color tw-pb-8-px tw-pt-16-px">Token Secret<app-mandatory-field [noSpacing]="true"></app-mandatory-field></p>
+              <input placeholder="Enter User Password" type="password" formControlName="tokenSecret" class="custom-focus tw-border tw-border-gray-300 tw-rounded tw-px-3 tw-py-2 tw-w-400-px" [ngClass]="[connectNetsuiteForm.controls.tokenSecret.invalid && connectNetsuiteForm.controls.tokenSecret.touched ? 'error-box' : 'normal-box']">
```
Comment (ashwin1111) [inline src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.html:31]: same

File: src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  connectNetsuiteForm: FormGroup;
+
+  ConfigurationCtaText = ConfigurationCta;
+
+  isOnboarding: boolean;
+
+  saveInProgress: boolean = false;
+
+  redirectLink: string = brandingKbArticles.onboardingArticles.INTACCT.CONNECTOR;
```
Comment (ashwin1111) [inline src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts:28]: use netsuite key

File: src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+  constructor(
+    private router: Router,
+    private formBuilder: FormBuilder,
+    private toastService: IntegrationsToastService,
+    private connectorService: NetsuiteConnectorService,
+    private mappingsService: NetsuiteMappingsService
+  ) { }
+
+  clearField() {
```
Comment (ashwin1111) [inline src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts:44]: private clearField(): void {

File: src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    const accountId = this.connectNetsuiteForm.value.accountId;
+    const tokenId = this.connectNetsuiteForm.value.tokenId;
+    const token_secret = this.connectNetsuiteForm.value.tokenSecret;
+
+    this.isLoading = true;
+    this.connectorService.connectNetsuite({
+      ns_account_id: accountId,
+      ns_token_id: tokenId,
+      ns_token_secret: token_secret
+    }).subscribe((response) => {
```
Comment (ashwin1111) [inline src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts:60]: use model for payload construction

**Comment by ashwin1111** [comment]
@coderabbitai review

File: src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+  private clearField() {
+    this.connectNetsuiteForm.get("accountId")?.setValue('');
+    this.connectNetsuiteForm.get("tokenId")?.setValue('');
+    this.connectNetsuiteForm.get("tokenSecret")?.setValue('');
+  }
+
+  save() {
+
+    const connector_payload = NetsuiteConnectorModel.constructPayload(
```
Comment (ashwin1111) [inline src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts:53]: camel caps

File: src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      this.connectNetsuiteForm.value.accountId,
```
Comment (ashwin1111) [inline src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts:54]: just send form, consistency

File: src/app/core/models/enum/enum.model.ts [hunk 7]
```diff
@@ -160,6 +160,7 @@ export enum QBOOnboardingState {
 
 export enum NetsuiteOnboardingState {
   CONNECTION = 'CONNECTION',
+  SUBSIDIARY_MAPPING = 'SUBSDIARY_MAPPING',
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:182]: don't find this in DB, can you take care of API changes too? might be a miss earlier

Comment (NileshPant1999) [reply src/app/core/models/enum/enum.model.ts:182]: it is there in code @ashwin1111 
but the name is `SUBSIDIARY`, changing the name

File: src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts [hunk 8]
```diff
@@ -50,14 +50,10 @@ export class NetsuiteConnectorComponent implements OnInit {
 
   save() {
 
-    const connector_payload = NetsuiteConnectorModel.constructPayload(
-      this.connectNetsuiteForm.value.accountId,
-      this.connectNetsuiteForm.value.tokenId,
-      this.connectNetsuiteForm.value.tokenSecret
-    );
+    const connectoPayload = NetsuiteConnectorModel.constructPayload(this.connectNetsuiteForm);
```
Comment (ashwin1111) [inline src/app/shared/components/netsuite/core/netsuite-connector/netsuite-connector.component.ts:53]: connectorPayload

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**